options-manager.js

```diff
@@ -191,10 +193,27 @@ function groupConfigs(paths, baseOptions, overrides) {
 	return arr;
 }
 
+function getIgnores(opts) {
+	const gitignores = globby.sync('**/.gitignore');
```

Comment:
It should only look for `.gitignore` files in not already ignored locations. So you need to pass the existing ignores here (in the `ignore` option). Right now it also looks in for example the `node_modules` folder, which is very slow.

Should also be a test for this (API test).